core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -164,13 +158,40 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
   }
 }
 
-object SmartTextVectorizer {
+object SmartTextVectorizer extends CleanTextFun {
   val MaxCardinality: Int = 100
   val MinTextLengthStdDev: Double = 0
+
   private[op] def partition[T: ClassTag](input: Array[T], condition: Array[Boolean]): (Array[T], Array[T]) = {
     val all = input.zip(condition)
     (all.collect { case (item, true) => item }, all.collect { case (item, false) => item })
   }
+
+  /**
+   * Computes a TextStats instance from a Text entry
+   *
+   * @param text            Text value (eg. entry in a dataframe)
+   * @param shouldCleanText Whether or not the text should be cleaned
+   * @param maxCardinality  Max cardinality to keep track of in maps (relevant for the text length distribution here)
+   * @tparam T              Feature type that the text value is coming from
+   * @return                TextStats instance with value and length counts filled out appropriately
+   */
+  private[op] def computeTextStats[T <: Text : TypeTag](
+    text: T#Value,
+    shouldCleanText: Boolean,
+    maxCardinality: Int
+  ): TextStats = {
+    // Go through each token in text and start appending it to a TextStats instance
+    val lengthsMap: Map[Int, Long] = TextTokenizer.tokenizeStringOpt(text).tokens.value
+      .foldLeft(Map.empty[Int, Long])(
+        (acc, el) => TextStats.additionHelper(acc, Map(el.length -> 1L), maxCardinality)
+      )
+    val (valueCounts, lengthCounts) = text match {
```

Comment:
when we reach `RawFeatureFilter.MaxCardinality` for `valueCounts`, will `lengthCounts` also stop accumulating ?

Reply:
I think in my earlier attempt to track token length distribution i also ran into this issue...

Reply:
nvm, this is taken care of by `val newLengthCounts = additionHelper(l.lengthCounts, r.lengthCounts, maxCardinality)` , pls disregard this comment :D